src/lib/sql/sql_translator.cpp

```diff
@@ -782,6 +794,18 @@ void SQLTranslator::_translate_select_list_groupby_having(const hsql::SelectStat
     if (hsql_expr->type == hsql::kExprStar) {
       AssertInput(_from_clause_result, "Can't SELECT with wildcards since there are no FROM tables specified");
 
+      if (is_aggregate) {
```

Comment:
This is unclear to me. You iterate over all pre_aggregate_expressions, which are the groupby expressions and arguments to aggregates. Then you assert that all of these are among the groupby columns. 

What about `SELECT * FROM x GROUP BY a, b HAVING SUM(x+y) > 5`? Wouldn't you needlessly require `x+y` to be part of the groupby expressions, here?
What about table-specific wildcards like in `SELECT t1.*, t2.a, SUM( t2.b) FROM t1, t2 GROUP BY t1.a, t1.b, t2.a`? Why would ` t2.b` need to be part of the groupby expressions?
Please add a bit of testing regarding these two scenarios.

Reply:
> What about SELECT * FROM x GROUP BY a, b HAVING SUM(x+y) > 5? Wouldn't you needlessly require x+y to be part of the groupby expressions, here?

Assuming you mean `SUM(a+b)`, that still works because `a+b` is not part of the `pre_aggregate_lqp`.

We do not have any HAVING tests, so I will add two.

>  What about table-specific wildcards like in SELECT t1.*, t2.a, SUM( t2.b) FROM t1, t2 GROUP BY t1.a, t1.b, t2.a? Why would t2.b need to be part of the groupby expressions?

Optimally, I would like the computer to blow up into the user who comes up with that query. :(

Fixed and added tests.